### ISSUE TITLE
Y24-204 - GBS pipeline - enhancement - prevent Perform PCR button showing until state is 'started'

### ISSUE THREAD

**Issue by andrewsparkes** [sanger/limber#1795]
**User story**
As Scott T I would like the GBS pipeline to prevent the 'Perform PCR' button on the PCR2 plate which changes the plate state to 'passed' from appearing until after the user has done the 'bravo GBS PCR1 to GBS PCR2' bed verification which changes the plate state to 'started'. At the moment the 'Perform PCR' button is visible earlier and can be pressed to bypass the need to use the robot.

**Who are the primary contacts for this story**
Scott T

**Who is the nominated tester for UAT**
Scott T

**Acceptance criteria**
To be considered successful the solution must allow:
- [x] after adding tags on creation of the PCR2 plate, only the 'bravo GBS PCR1 to GBS PCR2' bed verification button should be visible, not the 'Perform PCR' button
- [x] once the robot step is completed and plate state is 'started', the 'Perform PCR' button should be visible
- [x] the 'Perform PCR' button should change the plate state to 'passed', and the 'Add a GBS PCR2 Pool Stock Tube' button should be visible

**Dependencies**
None

References
None

**Additional context**
Noticed whilst testing the ANOSPP pipeline that feeds into GBS.
Not sure how possible this is. The 'Perform PCR' button is the equivalent of the 'Manual Transfer' button that is there in case the robot is unavailable. See `app/models/presenters/minimal_pcr_plate_presenter.rb`. Also see the integration suite `spec/limber/gbs_pipeline_spec.rb`.

See integration suite test PR:
https://gitlab.internal.sanger.ac.uk/psd/integration-suite/-/merge_requests/268

**Comment by SujitDey2022** [comment]
Team Discussion 24 July 2024:

1. Need to confirm functionality of 'Perform PCR' button v/s Manual Transfer button.

config/locales/en.yml
    pcr:
      transitions:
        passed: 'Perform PCR'
      
2. Need to review the scope and user interaction benefits.

**Comment by SujitDey2022** [comment]
Hi @ScottThurston, I wanted to check if this request is still valid?

**Comment by ScottThurston** [comment]
Yes, I believe this is still a valid request in that we would still like this feature. 

I tried to test whether the PCR button still appears for GBS PCR2 plates in 'pending' state in UAT, but I got a 500 error, so I was unable to test it. 

<img width="1430" alt="Image" src="https://github.com/user-attachments/assets/59ee2d56-c4c9-4e17-ba17-206d7a3c83a7" />

<img width="557" alt="Image" src="https://github.com/user-attachments/assets/545b252d-db84-492a-9cee-7096a8a02b22" />

**Comment by SujitDey2022** [comment]
Need to see if this is working as Scott was unable to test it last time due to the 500 error

**Comment by andrewsparkes** [comment]
In the `config/locales/en.yml` file we have:

```
en:
  state_machine:
    default:
      transitions:
        passed: 'Manual Transfer'
        started: 'Manual Transfer'
        qc_complete: 'Mark QC Complete'
      transitioning:
        passed: 'Transferring...'
        started: 'Transferring...'
        qc_complete: 'Completing QC...'
    pcr:
      transitions:
        passed: 'Perform PCR'
      transitioning:
        passed: 'Performing PCR'

```

The GBS PCR2 plate uses the `Presenters::MinimalPcrPlatePresenter` presenter. This presenter sets the `state_transition_name_scope`: 
`self.state_transition_name_scope = :pcr`

In the view `app/views/labware/_simple_state_change.html.erb` we have the following which uses the `state_transition_name_scope`:

```
<% presenter.default_state_change do |transition| %>
  <%= form_for(presenter.labware) do |f| %>
    <%= hidden_field_tag :purpose_uuid, presenter.purpose.uuid %>
    <%= hidden_field_tag :labware_barcode, presenter.human_barcode %>
    <%= hidden_field_tag :state, transition.to %>
    <%= hidden_field transition.to, :reason, value: t(transition.to, scope: [:state_machine, presenter.state_transition_name_scope, :transitions]) %>
    <div class="d-grid gap-2">
    <%= f.submit t(transition.to, scope: [:state_machine, presenter.state_transition_name_scope, :transitions]),
                 class: 'btn btn-lg btn-warning',
                 data: { disable_with: t(transition.to, scope: [:state_machine, presenter.state_transition_name_scope, :transitioning]) } %>
    </div>
  <% end %>
<% end %>
```

The 'Perform PCR' button is doing the same as 'Manual transfer', transitioning the plate state.


<img width="1292" height="665" alt="Image" src="https://github.com/user-attachments/assets/e65d9265-7724-4839-8a2a-19d02828a920" />
